lib/mayaUsd/ufe/UsdStageMap.cpp

```diff
@@ -92,8 +93,10 @@ void UsdStageMap::addItem(const Ufe::Path& path, UsdStageWeakPtr stage)
 	fStageToObject[stage] = proxyShape;
 }
 
-UsdStageWeakPtr UsdStageMap::stage(const Ufe::Path& path) const
+UsdStageWeakPtr UsdStageMap::stage(const Ufe::Path& path)
 {
+	rebuildIfDirty();
```

Comment:
All the query methods on the stage map, will now rebuild first if the map is flagged as dirty.